app/controllers/api/v1/base_controller.rb

```diff
@@ -8,33 +8,14 @@ class BaseController < ApplicationController
       private
 
       def require_authentication!
-        authenticator = Authenticator.new(
-          user: User.find_by(id: request.headers["X-User-Id"])
-        )
+        user = User.find_by(id: request.headers["X-User-Id"])
 
-        if authenticator.valid_token?(request.headers["Authorization"])
-          sign_in authenticator.user
+        if user&.valid_token?(request.headers["Authorization"])
+          sign_in user
         else
           head :unauthorized
         end
       end
-
-      class Authenticator
-        attr_reader :user
-
-        def initialize(user:)
-          @user = user
-        end
```

Comment:
When a class only takes one constructor argument, that's a good sign that the method(s) should be on that class.
(Unless your class really should be a decorator, which I don't think is true here.)